src/Build/Evaluation/Evaluator.cs

```diff
@@ -604,11 +604,23 @@ private static ProjectTargetInstance ReadNewTargetElement(ProjectTargetElement t
         /// </summary>
         private void Evaluate(ILoggingService loggingService, BuildEventContext buildEventContext)
         {
-            string projectFile;
+            ErrorUtilities.VerifyThrow(_data.EvaluationId == BuildEventContext.InvalidEvaluationId, "There is no prior evaluation ID. The evaluator data needs to be reset at this point");
+
+            string projectFile = String.IsNullOrEmpty(_projectRootElement.ProjectFileLocation.File) ? "(null)" : _projectRootElement.ProjectFileLocation.File;
```

Comment:
I would add this block of code back inside the evaluation profile's using block. Otherwise evaluation profiling would miss the time spent in this. It's probably not much but hey, who knows what the future holds :)